### PR TITLE
Implement GetGATTChildren

### PR DESCRIPTION
**Pull request by dati91**
Implement the [GetGATTChildren](https://webbluetoothcg.github.io/web-bluetooth/#getgattchildren) function for webbluetooth.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

- [X] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14593)

### CODE REVIEW

**Comment by dati91** [comment]
Note: it depends on https://github.com/servo/servo/pull/14567 and also needs a rebase.

**Comment by jdm** [comment]
What does WIP in the title mean?

**Comment by dati91** [comment]
work in progress, maybe i should have used [do not merge yet] or something
I just wanted to indicate somehow that the first patch doesn't belong here, just a placeholder before the other pr lands.

**Comment by jdm** [comment]
We can use the S-blocked-on-external label for that. WIP usually means that it's not necessarily ready for review yet.

File: components/bluetooth/lib.rs
Comment (jdm) [inline components/bluetooth/lib.rs]: Let's add the https://webbluetoothcg.github.io/web-bluetooth/#getgattchildren link back.

File: components/script/dom/bluetooth.rs
Comment (jdm) [inline components/script/dom/bluetooth.rs]:
```rust
let result_uuid = if let Some(u) = uuid {
    let canonicalized = match uuid_canonicalizer(u) {
        Ok(canonicalized_uuid) => canonicalized_uuid.to_string(),
        Err(e) => {
            ...
        }
    };
    if uuid_is_blocklisted(canonicalized.as_ref(), Blocklist::All) {
        ...
    }
    Some(canonicalized)
} else {
    None
};

**Review by jdm — CHANGES_REQUESTED** [review]
This is a nice improvement that reduces code complexity.

**Comment by jdm** [comment]
@bors-servo: delegate+
Squash and r=jdm.

**Pull request merged**
